### ISSUE TITLE
Fabric Manager unable to start

### ISSUE THREAD

**Issue by Hexoplon** [siderolabs/extensions#511]
Using the default config, and the Nvidia production drivers, the fabric manager extension keeps crashing. It exits, with an error compalining that `FABRIC_NODE_CONFIG_FILE` is invalid. 

What I've tried:
* Recompiling the extension to add updated topology files
* Updating to a newer release of the fabric manager
  * This yielded a different result, instead saying that the "DATABASE_PATH" variable had an invalid value. However, DATABASE_PATH is not set in the default config. When adding this variable, the config once again becomes invalid

Tested on a HPE Cray XD670 with 8xH200.

Also noticed that topology files for H100 and H200 cards are not added to the fabric manager extension, will add a separate PR to address this.

**Comment by smira** [comment]
Please provide some extension logs for us to understand the problem better, thank you!

**Comment by Hexoplon** [comment]
@smira The log is not very helpful, it just repeats this one line

`<node ip>: fabric manager config file item: FABRIC_NODE_CONFIG_FILE has an invalid value`.

This is with the default fabric manager config file, as included in the extension. Can't really tell what would be wrong, unless there is an issue with one of the paths and the error message just is misleading.

**Comment by askedrelic** [comment]
Very timely, I've just been investigating the same issue myself with a 8 H100 GPU machine.

I believe, but haven't yet confirmed, that this bug is due to the Talos fabricmanager extension not including all topology files in the fabricmanager release. This commit is my fix, that I'm still trying to test locally: https://github.com/siderolabs/extensions/commit/932f4d692f62b9557c828c31562c25f71279a294

For example, on my talos machine:
```
12:40:09 ❯ talosctl list -lH /usr/local/share/nvidia/nvswitch/
NODE             MODE         UID   GID   SIZE(B)   LASTMOD      LABEL   NAME
147.185.40.246   drwxr-xr-x   0     0     98 B      2 days ago           .
147.185.40.246   -rw-r--r--   0     0     266 kB    2 days ago           dgx2_hgx2_topology
147.185.40.246   -rw-r--r--   0     0     618 kB    2 days ago           dgxa100_hgxa100_topology
147.185.40.246   -rw-r--r--   0     0     12 kB     2 days ago           fabricmanager.cfg
```

And now locally:
```
$ gmake local-nvidia-fabricmanager-production PLATFORM=linux/amd64 DEST=_out
13:36:41 ❯ ls -l _out/rootfs/usr/local/share/nvidia/nvswitch
total 23840
-rw-r--r--@ 1 mbehrens  staff   260K Jan 20  2022 dgx2_hgx2_topology
-rw-r--r--@ 1 mbehrens  staff   604K Jan 20  2022 dgxa100_hgxa100_topology
-rw-r--r--@ 1 mbehrens  staff   4.0K Jan 20  2022 dgxgh200_hgxgh200_16gpus_osfp_cable_connections.csv
-rw-r--r--@ 1 mbehrens  staff   4.4K Jan 20  2022 dgxgh200_hgxgh200_16gpus_osfp_connections.csv
-rw-r--r--@ 1 mbehrens  staff   1.2M Jan 20  2022 dgxgh200_hgxgh200_16gpus_topology
-rw-r--r--@ 1 mbehrens  staff    11K Jan 20  2022 dgxgh200_hgxgh200_16gpus_trunk_connections.csv
-rw-r--r--@ 1 mbehrens  staff   8.0K Jan 20  2022 dgxgh200_hgxgh200_32gpus_osfp_cable_connections.csv
-rw-r--r--@ 1 mbehrens  staff   8.7K Jan 20  2022 dgxgh200_hgxgh200_32gpus_osfp_connections.csv
-rw-r--r--@ 1 mbehrens  staff   4.8M Jan 20  2022 dgxgh200_hgxgh200_32gpus_topology
-rw-r--r--@ 1 mbehrens  staff    22K Jan 20  2022 dgxgh200_hgxgh200_32gpus_trunk_connections.csv
-rw-r--r--@ 1 mbehrens  staff   198K Jan 20  2022 dgxgh200_hgxgh200_8gpus_topology
-rw-r--r--@ 1 mbehrens  staff   123K Jan 20  2022 dgxh100_hgxh100_topology
-rw-r--r--@ 1 mbehrens  staff    55K Jan 20  2022 dgxh800_hgxh800_topology
-rw-r--r--@ 1 mbehrens  staff    12K Jan 20  2022 fabricmanager.cfg
-rw-r--r--@ 1 mbehrens  staff   3.0M Jan 20  2022 oberon_gh_32gpus_topology
```

**Comment by askedrelic** [comment]
I was able to get the nvidia-fabricmanager-lts version working, by manually rebuilding the extension to include the topology fix and running with the related lts nvidia-container-toolkit and nvidia-open-gpu-kernel-modules. This would point the bug more toward nvidia-fabricmanager. Perhaps it would be worth filing to NVIDIA.

I also tried re-building fabricmanager, container-toolkit and open-gpu-kernel modules with the next fabricmanager version 550.127.05 and the same fabricmanager error still occurred.

**Comment by Hexoplon** [comment]
Yeah, I'm gonna try with the 550.127.05 driver and fabric manager versions as well, but if it does not work for you it will probably not work here either. I am not able to use the LTS versions of the driver release either, as the H200 cards are to new for that driver. Will have to see if I can get it working in some other way.

**Comment by Hexoplon** [comment]
@askedrelic Still no luck on my part. Have you enabled any specific kernel modules, which are not listed in the documentation, to get it to work?

**Comment by askedrelic** [comment]
Hey @Hexoplon I stopped investigating since the nvidia-fabricmanager-lts version was able to unblock us. I would like to use latest, but can wait until someone from Talos takes a look.

For more data, our working Ubuntu server is 22.04 with kernel 5.15.0-126-generic and nvidia drivers/fabricmanager 550.127.05.

I was thinking, is there was a way to get a shell inside the container trying to run fabricmanager? Being able to debug realtime would be quicker vs having to re-build the extensions and re-install everytime. It's still hard for me see if this a Talos permissions issue or an NVIDIA issue.

**Comment by Hexoplon** [comment]
I've also got it working just fine on a RHEL 8 server. Even afte rmodifying the config file to match the replacements that Talos are performing in the extension, and it runs just fine there. When I diff the talos config file, with the one from the RHEL server, they are identical. Yet, somehow fabric manager on Talos says the config file is invalid.

Perhaps the issue is not directly related to the config file, but rather a missing host mount or perhaps missing kernel modules.

**Comment by frezbo** [comment]
We'll try to look into this and fix before the 1.9 release

**Comment by Hexoplon** [comment]
@frezbo fantastic! Let me know if I can be of any assistance in testing

**Comment by frezbo** [comment]
> @frezbo fantastic! Let me know if I can be of any assistance in testing

actually if you could post the output of `talosctl logs syslogd` it could be helpful, fabricmanager logs to syslog

**Comment by Hexoplon** [comment]
@frezbo the only entry in the syslogd log is from nvidia-persistenced:  
`<node ip>: {"content":"nvidia-persistenced: Started (14)","facility":3,"hostname":"localhost","priority":29,"severity":5,"tag":"unknown","timestamp":"2024-11-19T20:57:15Z"}`
